skyportal/handlers/comment.py

```diff
@@ -135,11 +135,20 @@ def delete(self, comment_id):
               application/json:
                 schema: Success
         """
-        # TODO: Check ownership
-        comment = Comment.query.get(comment_id)
-        DBSession().delete(comment)
-        DBSession().commit()
-
+        user = (self.current_user.username if hasattr(self.current_user, 'username') else self.current_user.name)
+        roles = (self.current_user.roles if hasattr(self.current_user, 'roles') else '')
+        c = Comment.query.get(comment_id)
+        source_id = c.source_id
+        author = c.author
+        if ("Super admin" in [role.id for role in roles]) or (user == author):
+            try:
+                Comment.query.filter_by(id=comment_id).delete()
+                DBSession().commit()
+            except ValidationError as e:
+                return self.error('Invalid/missing parameters: '
```

Comment:
What does this error message mean?

Reply:
Hmm, I think the original author must've copied this from somewhere else, but it's not really applicable here. I'll update it.